### PR TITLE
Accept encrypted messages that use hcp_minimal

### PR DESCRIPTION
**Pull request by dkg**
in draft-ietf-lamps-header-protection-22, hcp_minimal recommends "[...]" as the obscured Subject header.  In the pending draft -23 (hopefully released this week, going into a working group last call), the same HCP will be renamed to hcp_baseline, but it still recommends the use of "[...]" for the obscured Subject header.

### CODE REVIEW

**Comment by hpk42** [comment]
sounds good -- could you add a test similar to https://github.com/deltachat/chatmail/blob/main/chatmaild/src/chatmaild/tests/test_filtermail.py#L56 ? 
It's probably best if you modify https://github.com/deltachat/chatmail/blob/main/chatmaild/src/chatmaild/tests/mail-data/encrypted.eml#L3  to take the parameter from the test, and run the test with these two subjects. 

As to running (offline) tests there is no documented way, but here is the flow: 
```
scripts/initenv.sh
source venv/bin/activate
pytest chatmaild
```

this should run any of the abovementioned tests. Some will be skipped if you have no root-level "chatmail.ini" but that's fine. 
- As to running the tests for "cmdeploy test" you would need a chatmail.ini 
Let us know if you have troubles setting up a "pytest chatmaild" run to verify things locally.

**Review by hpk42 — APPROVED** [review]
thanks, lgtm.

**Comment by hpk42** [comment]
tests fail though :) (it's the CI / isolated chatmaild tests) that are relevant.  Did this pass for you locally?

File: chatmaild/src/chatmaild/tests/test_filtermail.py
Comment (hpk42) [inline chatmaild/src/chatmaild/tests/test_filtermail.py:61]: for this call to work you need to modify https://github.com/deltachat/chatmail/blob/main/chatmaild/src/chatmaild/tests/plugin.py#L74 as well.  arguably we could use `**kwargs` but that's not the case currently.

Comment (hpk42) [reply chatmaild/src/chatmaild/tests/test_filtermail.py:61]: i'd probably let `maildata()` accept an optional `subject` parameter, and pass it on to the format() call  (formatting is IIRC fine with extra parameters that are not used in a template-string).

Comment (hpk42) [reply chatmaild/src/chatmaild/tests/test_filtermail.py:61]: please do try to run "pytest chatmaild" locally -- this gives more direct feedback compared to CI.

Comment (dkg) [reply chatmaild/src/chatmaild/tests/test_filtermail.py:61]: the tests didn't pass for me locally (i understood your recommendation to test but didn't have an environment set up that i felt comfortable running the venv you described).  I'm working on sorting that out.

Comment (hpk42) [reply chatmaild/src/chatmaild/tests/test_filtermail.py:61]: truth be told we so far only cared for ./scripts/*.sh invocations -- so far nobody tried to land a PR to chatmaild except the three core maintainers :)
we are also using "black" and "ruff" for formatting.  This is done with "cmdeploy fmt" or "venv/bin/cmdeploy fmt" if you use the environment created by `./scripts/initenv.sh`

**Comment by dkg** [comment]
the tests pass locally for me now, i don't understand why i'm not seeing the two failing CI builds above, but i think i can fix them anyway.

File: chatmaild/src/chatmaild/tests/plugin.py
Comment (hpk42) [inline chatmaild/src/chatmaild/tests/plugin.py:74]: i suggest to be more conservative here: 
```suggestion
    def maildata(name, from_addr, to_addr,  **kwargs):
```

**Comment by dkg** [comment]
just pushed a final complaint from the linter

**Comment by hpk42** [comment]
code is good but "ruff chatmaild" complains.

**Comment by hpk42** [comment]
image

**Comment by dkg** [comment]
I'm reading the CI reports, thanks :)  you don't need to cc them here :)

**Comment by hpk42** [comment]
ah, can you try something?  
if you install "tox" (io assumed it's packaged for debian) and then run "cd chatmaild ; tox"?

**Comment by hpk42** [comment]
> I'm reading the CI reports, thanks :) you don't need to cc them here :)

ah, i thought you couldn't see them.

**Comment by dkg** [comment]
some of these CI failures appear to be flaky things that are not under my control:

20240713_12h27m49s_grim

**Comment by hpk42** [comment]
yip, your PR is not allowed to access some secrets used for live-deploying and testing to staging2.testrun.org 
in invited you as outside collaborator. then it all works, and CI-runs don't need maintainer approval.

**Comment by dkg** [comment]
so, i accepted your request, and then triggered the job to re-run, and it still seems to fail in the same way.   :shrug: I'll leave this in your capable hands ☺

**Comment by hpk42** [comment]
k, maybe @nami or @link2xt can look into the precise github-rules there, i am less involved with github/CI magic myself. 
thanks for caring and congrats on a merged PR :)

**Pull request merged**
